src/include/stir/listmode/CListRecordSAFIR.h

```diff
@@ -0,0 +1,237 @@
+/* CListRecordSAFIR.h
+
+ Coincidence Event Class for SAFIR: Header File
+ Jannis Fischer
+ jannis.fischer@cern.ch
+
+	Copyright 2015 ETH Zurich, Institute of Particle Physics
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+ */
+
+/*!
+
+  \file
+  \ingroup listmode
+  \brief Declaration of class stir::CListEventSAFIR and stir::CListRecordSAFIR with supporting classes
+
+  \author Jannis Fischer
+*/
+
+#ifndef __local_SAFIR_CListRecordSAFIR_H__
+#define __local_SAFIR_CListRecordSAFIR_H__
+
+#include "stir/listmode/CListRecord.h"
+#include "stir/DetectionPositionPair.h"
+#include "stir/Succeeded.h"
+#include "stir/ByteOrder.h"
+#include "stir/ByteOrderDefine.h"
+
+#include "boost/make_shared.hpp"
+#include "boost/static_assert.hpp"
+#include "boost/cstdint.hpp"
+
+#include "stir/listmode/ListEventRecordMapFromFile.h"
+
+
+START_NAMESPACE_STIR
+
+/*!
+Provides interface of the record class to STIR by implementing get_LOR(). It uses a map from detector indices to coordinates to specify LORAs2Points from given detection pair indices.
+*/
+template <class Derived>
+class CListEventSAFIR : public CListEvent
+{
+public:
+	/*! Default constructor will not work as it does not initialize a map to relate
+	detector indices and space coordinates. Always use other constructor with a map pointer. Or use set_map( shared_ptr<ListEventRecordMapFromFile> new_map ) after default construction.
```

Comment:
as it doesn't work, do you need it? If you want to disable it, make it a private member (possibly unimplemented).